eng/tools/GenerateFiles/Directory.Build.targets.in

```diff
@@ -80,7 +80,8 @@
       <DefaultRuntimeFrameworkVersion Condition=" '$(IsServicingBuild)' != 'true' AND
           '%(TargetFramework)' == '${DefaultNetCoreTargetFramework}' AND
           '$(TargetLatestDotNetRuntime)' != 'false' ">${MicrosoftNETCoreAppRuntimeVersion}</DefaultRuntimeFrameworkVersion>
-      <RuntimePackRuntimeIdentifiers Condition=" '$(DotNetBuildSourceOnly)' == 'true' ">$(TargetRuntimeIdentifier)</RuntimePackRuntimeIdentifiers>
+      <!-- Workaround https://github.com/dotnet/sdk/pull/45487: The .NET SDK requires the crossgen2 host runtime identifier to be in the runtime pack runtime identifiers list. -->
+      <RuntimePackRuntimeIdentifiers Condition="'$(DotNetBuildOrchestrator)' == 'true'">$(NETCoreSdkRuntimeIdentifier);$(TargetRuntimeIdentifier)</RuntimePackRuntimeIdentifiers>
```

Comment:
Is it required that we conditionalized this on Orchestrator (or VMR switches at all?)

Reply:
We can probably change this to `DotNetBuild` instead of `DotNetBuildOrchestrator`.

We need to have `$(TargetRuntimeIdentifier)` in the list to support two-stage source-build. We need `$(NETCoreSdkRuntimeIdentifier)` here for the workaround, and in the crossgen2 list to support running crossgen2.

If we were to just append `$(TargetRuntimeIdentifier)` to the list, that would solve our builds; however, it would break the following scenario (which we don't have, but I could see a partner having at some point):

Cross-building from a source-built RID to a completely different source-build RID (like bootstrapping a new distro from an older source-built distro) within the VMR infra. In that case, we'd need to make sure to add the `$(NETCoreSdkRuntimeIdentifier)` to the list, in case it isn't present. I'll change this to use `DotNetBuild`.

Overwriting vs appending really comes down to this question: Do we want to have RIDs listed that we can't actually restore for, as long as we never try to restore for them? I think we're fine with this, so I'll change this to append.